sway-lib-std/src/message.sw

```diff
@@ -0,0 +1,47 @@
+library message;
+
+use ::outputs::{
+        Output,
+        output_count,
+        output_type,
+        output_amount,
+    };
+use ::revert::revert;
+
+/// Sends a message to `recipient` of length `msg_len` through `output` with amount of `coins`
+///
+/// # Arguments
+///
+/// * `coins` - Amount of base asset sent
+/// * `msg_len` - Length of message data, in bytes
+/// * `recipient` - The address of the message recipient
+// TODO: decide if `msg_len` can be determined programatically rather than passed as an arg
+pub fn send_message(coins: u64, msg_len: u64, recipient: b256) {
+    let mut index = 0;
+    let mut output_index = 0;
+    let mut output_found = false;
+
+    // If an output of type `Output::Message` is found, check if its `amount` is
+    // zero. As one cannot transfer zero coins to an output without a panic, a
+    // message output with a value of zero is by definition unused.
```

Comment:
Does this hold true for message outputs? (mostly compied from the `transfer_to_output` implementation where the `Output::Variable` type is used.

Reply:
From the spec: "this output type is unspendable and can be pruned from the UTXO set.". That tells me that if you actually manage to find an `OutputMessage`, then it's available regardless of the value of `amount`, but I could be wrong. I'll let @adlerjohn comment on this for the accurate answer.

Reply:
paging @adlerjohn ^^